packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -762,27 +768,32 @@ function areCompatibleNodes(client: Node, ssr: Node, vnode: VNode, renderer: Ren
 
     clientAttrsNames.forEach((attrName) => {
         if (getAttribute(client, attrName) !== getAttribute(ssr, attrName)) {
-            if (process.env.NODE_ENV !== 'production') {
-                logError(
-                    `Mismatch hydrating element <${getProperty(
-                        client,
-                        'tagName'
-                    ).toLowerCase()}>: attribute "${attrName}" has different values, expected "${getAttribute(
-                        client,
-                        attrName
-                    )}" but found "${getAttribute(ssr, attrName)}"`,
-                    vnode.owner
-                );
+            // Check if the root element attributes have expressions, if it does then we need to delegate hydration
+            // validation to haveCompatibleStaticParts.
+            // Note if there are no parts then it is a fully static fragment.
```

Comment:
I'd add a note here that, if there _is_ a part with `partId === 0`, then it will always be the first one. This is guaranteed by the compiler.

Reply:
Updated! - 51ef025cc8397ac0af8c6287df8ea5bb4a50e680